datalad/interface/tests/test_save.py

```diff
@@ -113,3 +115,28 @@ def test_recursive_save(path):
     newstates = [d.repo.get_hexsha() for d in (ds, subds, subsubds)]
     for old, new in zip(states, newstates):
         assert_not_equal(old, new)
+
+    ## now let's check saving "upwards"
+    assert not subds.repo.dirty
+    create_tree(subds.path, {"testnew": 'smth'})
+    indexed_files = subds.repo.get_indexed_files()
+    assert subds.repo.dirty
+    assert ds.repo.dirty
+
+    assert not subsubds.repo.dirty
+    create_tree(subsubds.path, {"testnew2": 'smth'})
+    assert subsubds.repo.dirty
+    # and indexed files didn't change
+    assert_equal(indexed_files, subds.repo.get_indexed_files())
+    subsubds.save(message="saving", super_datasets=True,
+                  auto_add_changes=True)
+    assert not subsubds.repo.dirty
+    # but its super should have got only the subsub saved
+    # not the file we created
+    assert subds.repo.dirty
```

Comment:
This does not test, what you want to be the result according to the comment. Just "dirty" is also true, if your `super_datasets=True` did nothing.